### PR TITLE
Create Indicator Relationships script

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:https://github.com/demisto/etc/issues/36050
## Description
A new script that creates relationships between indicators.

## Screenshots
image
image
image
image

## Video
https://user-images.githubusercontent.com/58938354/116858559-3b9fa980-ac07-11eb-9f1f-758ff4e1b7af.mov

# set of tests:

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.yml [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+outputs:
+script: '-'
+system: false
+tags:
+- basescript
+timeout: '0'
+type: python
+subtype: python3
+dockerimage: demisto/python3:3.9.4.18682
+fromversion: 6.1.0
```
Comment (bakatzir) [inline Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.yml:289]:
```suggestion
fromversion: 6.2.0
```

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+def validate_arguments(args: dict) -> Dict[str, str]:
+    """Get the args of the command and validate the arguments.
+
+    :return: raise an error if one of the validations fail.
+    :rtype: ``None``
+    """
+    if len(argToList(args.get('entity_a'))) > 1:
+        raise Exception("entity_a is a list, Please insert a single entity_a to create the relationship")
+    if len(argToList(args.get('entity_b_type'))) > 1:
+        raise Exception("entity_b_type is a list, Please insert a single type to create the relationship")
```
Comment (bakatzir) [inline Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py:112]: should we also add check for
`len(argToList(args.get('entity_a_type'))) > 1`?

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        relationships = create_relation_command_using_query(args)
+    else:
+        relationships = create_relationships_with_args(args)
+
+    if len(relationships) == 1:
+        human_readable = f"Relationship for {args.get('entity_a')} was created successfully."
+    elif len(relationships) > 1:
+        human_readable = f"Relationships for {args.get('entity_a')} were created successfully."
+    else:
+        human_readable = f"Relationships were not created for {args.get('entity_a')}."
```
Comment (bakatzir) [inline Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py:174]: please add the exact number of relationships created in the title.

**Review by bakatzir — COMMENTED** [review]
Good job using the `IndicatorsSearcher`.

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py (hunk 3 above)
Comment (orhovy) [reply Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py]: done

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py (hunk 2 above)
Comment (orhovy) [reply Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py:112]: sure, why not, added

**Review by bakatzir — APPROVED** [review]
I'm good with the additions, let's let @yuvalbenshalom review as well.

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+        # save only the value and type of each indicator
+        indicators.extend({'entity_b': ioc.get('value'), 'entity_b_type': ioc.get('indicator_type')}
+                          for ioc in fetched_indicators)
+        fetched_indicators = search_indicators.search_indicators_by_version(query=query, size=PAGE_SIZE).get('iocs')
+
+    return indicators
+
+
+def remove_existing_entity_b_indicators(args: dict) -> list:
```
Comment (yuvalbenshalom) [inline Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py:34]: it is unclear what this function dose and why it is being called.
consider changing name or add a description to it in the doc string.

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
good work.
please see my comment and consider changing  internal functions signature to take specific arguments and not all args, if possible.

**Comment by orhovy** [comment]
@ShirleyDenkberg @kirbles19 Please take a quick look at the descriptions of the script arguments, thanks.

File: Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py (hunk 4 above)
Comment (orhovy) [reply Packs/Base/Scripts/CreateIndicatorRelationship/CreateIndicatorRelationship.py]: added docstrings

**Comment by ShirleyDenkberg** [comment]
@yuvalbenshalom @bakatzir Doc review completed.
As I mentioned to @orhovy, the following predefined values are misspelled:
- attachement-of  should be attachment-of
- bundeled-in should be bundled-in
- embbeded-in should be embedded-in

**Comment by orhovy** [comment]
> @yuvalbenshalom @bakatzir Doc review completed.
> As I mentioned to @orhovy, the following predefined values are misspelled:
> 
> * attachement-of  should be attachment-of
> * bundeled-in should be bundled-in
> * embbeded-in should be embedded-in

@ShirleyDenkberg  Fixed

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**